.github/workflows/pylint.yml

```diff
@@ -0,0 +1,31 @@
+name: Code Format Check
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.9]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8
+          if [ -f requirements_dev.txt ]; then pip install -r requirements_dev.txt; fi
+
+      - name: Lint
+        run: |
+          flake8 --max-line-length=120 --ignore=F401,W503 --count --show-source --statistics orsopy
+          flake8 --max-line-length=120 --ignore=F401,W503 --count --show-source --statistics tests
```

Comment:
`black -l 120 orsopy tests`

Reply:
See above

Reply:
Yeah I think this would just change the code to black formatting and then not do anything... I am not sure how black works on CI though so it might fail (if it works like `yapf`

Reply:
Sorry, it should have the --check flag

Reply:
Then it sees if the formatting is correct.

Reply:
I agree - check but don't change the code (commit or push anything) in CI.  This all seems ok to me.

Reply:
The check will fail, even if the user followed the formatting guidelines as `isort `is run after `black`. This means that long `from ... import ....` statements will be split up on many lines by black. This was one of the reasons I liked running isort, it not only creates the same import orders for all files it also makes the format more readable than an import statement that runs of 10-20 lines.

95% of what black does is making code compatible with PEP 8, so code that is not formatted by black will likely fail `flake8`. I would therefore see the test as redundant. The possible existing formatting errors from someone not running `black` but passing `flake8` can only be minor.

Reply:
Here is how blacks diff would look like on the current branch (from isort version back to black version):
```diff
--- orsopy\fileio\base.py       2021-12-06 20:07:25.239734 +0000
+++ orsopy\fileio\base.py       2021-12-09 07:57:32.837696 +0000
@@ -10,12 +10,25 @@
 import warnings

 from collections.abc import Mapping
 from contextlib import contextmanager
 from copy import deepcopy
-from dataclasses import (_FIELD, _FIELD_INITVAR, _FIELDS, _HAS_DEFAULT_FACTORY, _POST_INIT_NAME, MISSING, _create_fn,
-                         _field_init, _init_param, _set_new_attribute, dataclass, field, fields)
+from dataclasses import (
+    _FIELD,
+    _FIELD_INITVAR,
+    _FIELDS,
+    _HAS_DEFAULT_FACTORY,
+    _POST_INIT_NAME,
+    MISSING,
+    _create_fn,
+    _field_init,
+    _init_param,
+    _set_new_attribute,
+    dataclass,
+    field,
+    fields,
+)
 from inspect import isclass
 from typing import Any, Generator, List, Optional, TextIO, Tuple, Union

 import numpy as np
 import yaml
```